150627362 create nag notifications

aenkya opened the pull request:
### What does this PR do?
Create Nag Notifications
### Description of Task to be completed?

* Add allowance to configure Nag Interval Times
* Notify user after nag time elapses for checks that are down
* Create a nag mode icon for the dashboard

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/n/projects/2099685/stories/150627362

Commit:
Add Nag Time Slider to Template

Add css styles and js scripts for the configuration of the nag time
Add description of the nag time to modal

Commit:
[Feature #150627362] Add Configuration of Nag time
- Modify Model to add nag_time, nag_mode
- Add nag_time and nag_mode to timeout form
- Modify my_checks.html and checks.js to include nag_interval slider
- Makemigrations and migrate

Commit:
[Feature #150627362] Receive Email When Nag Time has Elapsed
- Modify Checks model to add next_nag that captures time of next nag

Commit:
[Feature #150627362] Add Check on Nag Mode with Icon
- Add in_nag_period() which checks if check is in nag period
- Add Icon for nag mode

Commit:
[Feature #150627362] Write test for nagging
Test that next_nag is updated
Change nag icon colors

Commit:
[Feature #150627362] Reorgnize the Nag Time Display on the UI

Commit:
Merge branch 'development' into 150627362-create-nag-notifications

Commit:
[Feature 150627362] Create Nag Notifications

Commit:
[Feature 150627362] Create Nag Notifications

NgangaMaryanne commented on .gitignore:
```diff
@@ -3,6 +3,8 @@ __pycache__/
 hc.sqlite
 static-collected
+*.sublime-project
+*.sublime-workspace
```
why do we have these here?

NgangaMaryanne commented on hc/front/tests/test_update_timeout.py:
```diff
@@ -11,19 +11,21 @@ def setUp(self):
         self.assertRedirects(r, "/checks/")
 
         check = Check.objects.get(code=self.check.code)
+        print(check.nag_time)
```
This print statement is unnecessary. Please remove it.

NgangaMaryanne commented on hc/api/tests/test_sendalerts.py:
```diff
@@ -35,4 +36,17 @@ def test_it_handles_grace_period(self):
+        Command().handle_many()
+        checks = Check.objects.filter(status="down", nag_mode=True).first()
+        next_time = now + timedelta(minutes=3)
+        assert checks.next_nag >= next_time
```
previous asserts follow the convention `self.assert`. Please use the same convention for this nag. Also isnt the time for next nag supposed to be exactly 3 minute later?

Commit:
[ Feature 150627364] update-test-syntax-for-readability

Remove editor specific changes for sublime
Remove print debuggers
Unify assert syntax convention

Commit:
[ Feature 150627364] update-test-syntax-for-readability

Remove editor specific changes for sublime
Remove print debuggers
Unify assert syntax convention

Commit:
Merge branch 'development' into 150627362-create-nag-notifications

Commit:
Merge branch '150627362-create-nag-notifications' of https://github.com/andela/healthchecks-shockwave into 150627362-create-nag-notifications

Commit:
Merge branch 'development' into 150627362-create-nag-notifications
